src/vs/workbench/services/runtimeSession/common/runtimeSession.ts

```diff
@@ -447,14 +447,23 @@ export class RuntimeSessionService extends Disposable implements IRuntimeSession
 				workspaceFolder || undefined, configValue
 			);
 		} catch (error) {
-			this._logService.warn(`${NotebookSetting.workingDirectory}: Failed to resolve variables in '${configValue}':`, error);
+			this._logService.warn(`${NotebookSetting.workingDirectory}: Failed to resolve variables in '${configValue}'. Using default: '${defaultValue}'`, error);
 			return defaultValue;
 		}
 
 		// Check if the result is a directory that exists
-		if (await this.isValidDirectory(resolvedValue)) {
+		let resolvedValueUri: URI;
+		try {
+			resolvedValueUri = notebookUri.with({ path: resolvedValue });
```

Comment:
I put this in a try catch due to the [failure](https://d38p2avprg8il3.cloudfront.net/playwright-report-16997915866-24068/trace/index.html?trace=https://d38p2avprg8il3.cloudfront.net/playwright-report-16997915866-24068/data/400e3d608eec31dd5034b4f88fa42301b8760b85.zip) we saw once in e2e tests. I haven't been able to reproduce, but at least now if it happens again it won't fail to start the session.